### PR TITLE
Stage 2 Orchestrator RFC

### PR DESCRIPTION
**Pull request by ferozsalam**
Progress the Orchestrator RFC to Stage 2.

I have made an attempt to identify anything that will need updating if when the RFC process for this is complete, but I expect those with more familiarity with the entire Elastic stack will have a better idea!

Any additional concerns around the fieldset are also welcome.

### CODE REVIEW

File: rfcs/text/0012-orchestrator-field-set.md [hunk 1]
```diff
@@ -202,23 +202,30 @@ Examples of source data include:
...
+- Logstash should see no significant change.
+
+### Usage mechanisms
+
+- Elastic's detection-rules [repo][6] should see no change as there don't appear to
+  be any orchestrator-specific definitions in place.
+
+### ECS project
+
+Documentation updates might be required to reflect the new fieldset.
```
Comment (ferozsalam) [inline rfcs/text/0012-orchestrator-field-set.md:228]: @ebeahan I'm not sure if this is actually true or whether all necessary documentation will be auto-generated from the fieldset definitions.

Comment (ebeahan) [reply rfcs/text/0012-orchestrator-field-set.md:228]: @ferozsalam Great question - the additions to the ECS field reference docs will auto-generate from the fieldset definitions once the changes are added.

File: rfcs/text/0012-orchestrator-field-set.md [hunk 2]
```diff
@@ -202,23 +202,30 @@ Examples of source data include:
...
- * ECS project (e.g. docs, tooling)
-The goal here is to research and understand the impact of these changes on users in the community and development teams across Elastic. 2-5 sentences each.
--->
+As this RFC involves the creation of an entirely new fieldset, no breaking
+changes are envisaged. Some existing tooling might need updates to factor in the
+new fieldset's availability, however.
+
+### Ingestion mechanisms
+
+- The Filebeat [Kubernetes processor][5] will need updating, as it currently
```
Comment (ebeahan) [inline rfcs/text/0012-orchestrator-field-set.md:217]: @ChrsMark @jsoriano @kaiyan-sheng Any other integrations or components that might populate these orchestrator fields that we might want to capture here?

Comment (jsoriano) [reply rfcs/text/0012-orchestrator-field-set.md:217]: Same change will be needed in Cloud Foundry and Nomad features. Btw, I see that nomad docs are not available in the public guides, I will take a look.

Comment (jsoriano) [reply rfcs/text/0012-orchestrator-field-set.md:217]: Adding the missing docs for Nomad: https://github.com/elastic/beats/pull/24557

Comment (exekias) [reply rfcs/text/0012-orchestrator-field-set.md:217]: ++ I also expect Metricbeat Kubernetes module to need updates, in the line of the Filebeat processor/autodiscover provider: https://www.elastic.co/guide/en/beats/metricbeat/current/running-on-kubernetes.html

Comment (ferozsalam) [reply rfcs/text/0012-orchestrator-field-set.md]: Thanks all - I have added the mentioned services to the document.

**Review by exekias — APPROVED** [review]

**Comment by ebeahan** [comment]
Thanks all - will set the today's date and merge!

**Pull request merged**